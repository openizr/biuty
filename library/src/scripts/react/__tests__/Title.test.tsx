/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import UITitle from 'scripts/react/Title';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from 'react-dom';

let container = document.createElement('div');

describe('react/UITitle', () => {
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    ((container as unknown) as null) = null;
  });

  test('renders correctly - basic', () => {
    act(() => {
      render(<UITitle label="Test" level="1" modifiers="large" />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - modifier different than level', () => {
    act(() => {
      render(<UITitle label="Test" level="1" modifiers="5 large" />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    act(() => {
      render(<UITitle label="Test" id="test" />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with itemprop', () => {
    act(() => {
      render(<UITitle label="Test" itemProp="name" />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
