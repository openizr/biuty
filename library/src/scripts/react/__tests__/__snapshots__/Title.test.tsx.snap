// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/UITitle renders correctly - basic 1`] = `
<div>
  <h1
    class="ui-title ui-title--large--1"
  >
    Test
  </h1>
</div>
`;

exports[`react/UITitle renders correctly - modifier different than level 1`] = `
<div>
  <h1
    class="ui-title ui-title--5--large"
  >
    Test
  </h1>
</div>
`;

exports[`react/UITitle renders correctly - with id 1`] = `
<div>
  <h1
    class="ui-title ui-title--1"
    id="test"
  >
    Test
  </h1>
</div>
`;

exports[`react/UITitle renders correctly - with itemprop 1`] = `
<div>
  <h1
    class="ui-title ui-title--1"
    itemprop="name"
  >
    Test
  </h1>
</div>
`;
